confection/tests/test_config.py

```diff
@@ -580,32 +580,32 @@ def test_schedule():
 
     cfg = {"@schedules": "test_schedule.v2"}
     result = my_registry.resolve({"test": cfg})["test"]
-    assert isinstance(result, GeneratorType)
+    assert isinstance(result, Iterator)
```

Comment:
So the other somewhat controversial change is the removal of checks for the Generator type. Pydantic supports validating Python generators via the `Iterator` and `Iterable` types, but not through the `GeneratorType`. This is also how you're supposed to type stuff in Python so I think this should just be a documentation update but wanted to call it out.